### PR TITLE
aya-log: Make tokio optional

### PR DESCRIPTION
**Pull request by vadorovsky**
Use tokio only when `async_tokio` feature is enabled, otherwise use `std::thread`.

Tested on: https://github.com/vadorovsky/aya-examples/tree/main/notokio

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

**Review by dave-tucker — APPROVED** [review]
LGTM

File: aya-log/src/lib.rs [hunk 1]
```diff
@@ -63,14 +63,69 @@ use log::{error, Level, Log, Record};
...
+    let mut logs: PerfEventArray<_> = bpf.map_mut("AYA_LOGS")?.try_into()?;
+
+    for cpu_id in online_cpus().map_err(Error::InvalidOnlineCpu)? {
+        let mut buf = logs.open(cpu_id, None)?;
+
+        let log = logger.clone();
+        std::thread::spawn(move || {
+            let mut buffers = buffers();
+            loop {
+                let events = buf.read_events(&mut buffers).unwrap();
```
Comment (alessandrod) [inline aya-log/src/lib.rs:120]: this spin loops without sleeping

**Review by alessandrod — CHANGES_REQUESTED** [review]
This needs more work. As it is now it'll spin loop when using threads, using all the available CPUs :D 

We need to make this sleep, and in the process probably rework the API so that the caller can optionally read from its own threads (the threads are created from the caller - aya-log just provides an interface that can be polled)

**Comment by vadorovsky** [comment]
@alessandrod Thanks! I will open a new PR for that, since it's broken enough that doesn't make sense to rebase.

**Pull request closed without merging**